packages/util/src/signature.ts

```diff
@@ -86,12 +121,7 @@ export const toRpcSig = function (v: bigint, r: Buffer, s: Buffer, chainId?: big
  * NOTE: Accepts `v == 0 | v == 1` for EIP1559 transactions
  * @returns Signature
  */
-export const toCompactSig = function (v: bigint, r: Buffer, s: Buffer, chainId?: bigint): string {
-  const recovery = calculateSigRecovery(v, chainId)
-  if (!isValidSigRecovery(recovery)) {
-    throw new Error('Invalid signature v value')
-  }
```

Comment:
This removes the recovery check completely if I read the code correctly which shouldn't be the case I guess.